### ISSUE TITLE
False positive invalid-argument-type for TypeAlias built from a bare/unbound TypeVar

### ISSUE THREAD

**Issue by dgavrilov** [astral-sh/ty#4064]
### Summary

`ty` reports a false-positive `invalid-argument-type` when a parameter is annotated with a `TypeAlias` that is itself built from a bare `TypeVar` not tied to the enclosing function's generic scope (e.g. `SomeAlias: TypeAlias = "Generic[SomeTypeVar]"`, used as a plain, non-generic function's parameter type). A concrete `Generic[Concrete]` argument is rejected even though it satisfies the alias.

This is exactly the pattern used by `litestar.types.internal_types.TemplateConfigType` / `litestar.app.Litestar.__init__(..., template_config: TemplateConfigType | None = None, ...)`, so any Litestar app that passes a `TemplateConfig(...)` to `Litestar(template_config=...)` hits this. Both `mypy` and `pyright` accept the same code without complaint.

### Reproduction

```python
from __future__ import annotations

from dataclasses import dataclass, field
from typing import Generic, TypeAlias, TypeVar


class EngineProtocol:
    pass


class ConcreteEngine(EngineProtocol):
    pass


EngineType = TypeVar("EngineType", bound=EngineProtocol)


@dataclass
class Config(Generic[EngineType]):
    instance: EngineType | None = field(default=None)


# Alias built from a bare TypeVar not tied to any enclosing generic scope,
# mirroring litestar.types.internal_types.TemplateConfigType.
ConfigType: TypeAlias = "Config[EngineType]"


def register(config: ConfigType | None = None) -> None:
    pass


register(Config(instance=ConcreteEngine()))
```

```
$ uvx ty check repro.py
error[invalid-argument-type]: Argument to function `register` is incorrect
  --> repro.py:31:10
   |
31 | register(Config(instance=ConcreteEngine()))
   |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ Expected `Config[EngineType] | None`, found `Config[ConcreteEngine]`
   |
info: Function defined here
  --> repro.py:27:5
   |
27 | def register(config: ConfigType | None = None) -> None:
   |     ^^^^^^^^ -------------------------------- Parameter declared here
   |

Found 1 diagnostic
```

`mypy 2.3.0` and `pyright 1.1.411` both report no issues on the identical file.

### Real-world occurrence (Litestar)

```python
from litestar import Litestar
from litestar.contrib.jinja import JinjaTemplateEngine
from litestar.template.config import TemplateConfig

engine = JinjaTemplateEngine(directory="templates")
config = TemplateConfig(instance=engine)

app = Litestar(route_handlers=[], template_config=config)
```

```
error[invalid-argument-type]: Argument to `Litestar.__init__` is incorrect
  --> asgi.py:8:5
  |
8 |     template_config=config,
  |     ^^^^^^^^^^^^^^^^^^^^^^ Expected `TemplateConfig[EngineType] | None`, found `TemplateConfig[JinjaTemplateEngine]`
  |
info: Method defined here
  --> .venv/Lib/site-packages/litestar/app.py:173:9
```

(`litestar.types.internal_types.TemplateConfigType` is defined as `TemplateConfigType: TypeAlias = "TemplateConfig[EngineType]"`, the same shape as the minimal repro above.)

### Expected behavior

`Config[ConcreteEngine]` (resp. `TemplateConfig[JinjaTemplateEngine]`) should satisfy a parameter typed as the alias `Config[EngineType]` (resp. `TemplateConfig[EngineType]`), the same way `mypy` and `pyright` treat it — the free `EngineType` should be solved against the concrete argument rather than compared invariantly against the declared, unparameterized alias.

### Version

ty 0.0.62 (df75992e7 2026-07-21)
Python 3.14.6, Windows

**Comment by sharkdp** [comment]
Thank you for reporting this. Should be fixed by https://github.com/astral-sh/ruff/pull/27092
